R/eatModel.r

```diff
@@ -2552,60 +2555,67 @@ get.plausible <- function(file, quiet = FALSE, forConquestResults = FALSE)  {
                  return(PV)}}
 
 get.wle <- function(file)      {
+  # checks
+  checkmate::assert_character(file, len = 1)
             input <- eatTools::crop(scan(file, what = "character", sep = "\n", quiet = TRUE))
             input <- strsplit(input," +")
-            n.spalten <- max ( sapply(input,FUN=function(ii){ length(ii) }) )   
-            n.wle <- floor((n.spalten-1) / 4)                                   
+            n.spalten <- max ( sapply(input,FUN=function(ii){ length(ii) }) )
+            n.wle <- floor((n.spalten-1) / 4)
             input <- suppressWarnings(eatTools::asNumericIfPossible(data.frame( matrix( t( sapply(input,FUN=function(ii){ ii[1:n.spalten] }) ),length(input),byrow = FALSE), stringsAsFactors = FALSE), force.string = FALSE))
             valid <- na.omit(input)
             cat(paste("Found valid WLEs of ", nrow(valid)," person(s) for ", n.wle, " dimension(s).\n",sep=""))
             if (nrow(valid) != nrow(input)) { cat(paste("    ",nrow(input)-nrow(valid)," persons with missings on at least one latent dimension.\n",sep="")) }
             namen1<- c(rep ( x = c("n.solved", "n.total"), times = n.wle), rep ( x = c("wle", "std.wle"), times = n.wle))
-            namen2<- rep(rep ( paste(".", 1:n.wle, sep=""), each = 2),2)        
+            namen2<- rep(rep ( paste(".", 1:n.wle, sep=""), each = 2),2)
             colnames(valid)[(ncol(valid)-length(namen2)):1] <- c("ID","case")[1:(ncol(valid)-length(namen2))]
             colnames(valid)[(ncol(valid)-length(namen2)+1):ncol(valid)] <- paste(namen1,namen2,sep="")
             return(valid)}
 
 get.shw <- function(file, dif.term, split.dif = TRUE, abs.dif.bound = 0.6, sig.dif.bound = 0.3, p.value = 0.9) {
-            all.output <- list();   all.terms <- NULL                           
-            input.all <- scan(file,what="character",sep="\n",quiet=TRUE)        
+  # checks
+  checkmate::assert_character(file, len = 1)
+  checkmate::assert_character(dif.term, null.ok = TRUE)

```

Comment:
shall I check here, whether string in "file" contains "dif.term"?

Reply:
The `get.shw` function is not very well written, to be honest. See, for example the statement `ind <- grep("TERM",input.all)` which looks for all occurrences of `TERM` within the ASCII file. The result of `ind` must be a numerical vector with at least one element:
`if(length(ind)==0) {stop(paste("No TERM-statement found in file ",file,".\n",sep=""))}` could be replaced with `assert_numeric(ind, min.len = 1, unique=TRUE)`. However, if these check fails, a user message like `Assertion on 'ind' failed: Must have length >= 1, but has length 0.` is not very informative for the user, but I don't know whether checkmate allows to customize message wording. 

If `dif.term` is not missing, `dif.term` should occur in `input.all`. Maybe a statement like 
```
!missing(dif.term){
  ind2 <- grep(dif.term, input.all, fixed = TRUE)
  assert_numeric(ind2, len = 1) }
```
could be useful?

`fixed = TRUE` is necessary here as `dif.term` contains the conquest statement for differential item functioning (DIF) which ofen occurs special characters (asterisks). Without `fixed = TRUE`, statements like `item*gender` will not be found.